go/libraries/doltcore/sqle/dprocedures/dolt_pull.go

```diff
@@ -152,6 +152,10 @@ func doDoltPull(ctx *sql.Context, args []string) (int, int, error) {
 			if err != nil {
 				return noConflictsOrViolations, threeWayMerge, err
 			}
+			err = checkForUncommittedChanges(ctx, roots.Working, roots.Head)
```

Comment:
What's stopping us from supporting pull with dirty workspaces? It's no different from merge. Is this to comply with existing tests? Allowing it would be closer to git behavior and consistent with a pull being a fetch -> merge. Let's leave for now, and fast follow as priority allows.